Source/MLX/Ops.swift

```diff
@@ -819,10 +820,47 @@ public func loadArrays(url: URL, stream: StreamOrDevice = .default) throws -> [S
 
     switch url.pathExtension {
     case "safetensors":
-        let mlx_map = mlx_load_safetensors(filename, stream.ctx)!
-        defer { mlx_free(mlx_map) }
+        let mlx_safetensors = mlx_load_safetensors(filename, stream.ctx)!
+        defer { mlx_free(mlx_safetensors) }
 
-        return mlx_map_values(mlx_map)
+        let mlx_arrays = mlx_safetensors_data(mlx_safetensors)!
+        defer { mlx_free(mlx_arrays) }
+
+        return mlx_map_array_values(mlx_arrays)
+    default:
+        throw LoadSaveError.unknownExtension(url.pathExtension)
+    }
+}
+
+/// Load dictionary of ``MLXArray`` and metadata `[String:String]` from a `safetensors` file.
+///
+/// - Parameters:
+///     - url: URL of file to load
+///     - stream: stream or device to evaluate on
+///
+/// ### See Also
+/// - ``loadArrays(url:stream:)``
+/// - ``loadArray(url:stream:)``
+public func loadArraysAndMetadata(url: URL, stream: StreamOrDevice = .default) throws -> (
+    [String: MLXArray], [String: String]
+) {
```

Comment:
Here is a variant that also returns the metadata.  Not added to previous one as it will break API (return value)